### ISSUE TITLE
Enable api

### ISSUE THREAD

**Issue by liexes** [diamond-rip/Eden#408]
Please add api of eden, a owner of the plugin "Parties" "alessiopd" need the plugin api to add compatibility in the plugin.

In short, he needs the eden api to be able to make his party plugin compatible with eden, I hope you enable it for him to add this...

**Comment by GoodestEnglish** [comment]
I need to know what feature do he need. Every party stuff can be access in here: https://github.com/diamond-rip/Eden/blob/master/src/main/java/rip/diamond/practice/party/Party.java

**Comment by liexes** [comment]
> I need to know what feature do he need. Every party stuff can be access in here: https://github.com/diamond-rip/Eden/blob/master/src/main/java/rip/diamond/practice/party/Party.java

This not is a api

**Comment by liexes** [comment]
example need:

https://github.com/PikaMug/Unite/wiki/Getting-started#developers

**Comment by GoodestEnglish** [comment]
By what I mean, is right now he should access the party function by directly access to the party class. There's no point to create an extra API class when everything is already in Party.java

**Comment by AlessioDP** [comment]
Hello, what @liexes was asking to me on Discord was:
He wants to use Parties party system inside Eden plugin, but this is not possible because Eden is using its own system and its not extendible for others one.

The Unite example by PikaMug shows how is possible to use an interface to handle Party stuff and make it extendible for other implementations like your own party system or Parties one.
This would make your plugin party system a handler for others one.

Example:
When your plugin wants to get the party of a player, you may do something like getPartyOfPlayer(x), and gets your own Party class (and I cannot handle that one). You may return an interface for the Party itself, so I can return my Parties implementation of that interface and it would works with your plugin too.

This would allows other plugins to hooks and handles parties for your plugin, otherwise directly hooks into other party plugins or Unite (a bridge for some party plugins).

**Comment by liexes** [comment]
> Hello, what @liexes was asking to me on Discord was: He wants to use Parties party system inside Eden plugin, but this is not possible because Eden is using its own system and its not extendible for others one.
> 
> The Unite example by PikaMug shows how is possible to use an interface to handle Party stuff and make it extendible for other implementations like your own party system or Parties one. This would make your plugin party system a handler for others one.
> 
> Example: When your plugin wants to get the party of a player, you may do something like getPartyOfPlayer(x), and gets your own Party class (and I cannot handle that one). You may return an interface for the Party itself, so I can return my Parties implementation of that interface and it would works with your plugin too.
> 
> This would allows other plugins to hooks and handles parties for your plugin, otherwise directly hooks into other party plugins or Unite (a bridge for some party plugins).

@GoodestEnglish

**Comment by GoodestEnglish** [comment]
> Hello, what @liexes was asking to me on Discord was: He wants to use Parties party system inside Eden plugin, but this is not possible because Eden is using its own system and its not extendible for others one.
> 
> The Unite example by PikaMug shows how is possible to use an interface to handle Party stuff and make it extendible for other implementations like your own party system or Parties one. This would make your plugin party system a handler for others one.
> 
> Example: When your plugin wants to get the party of a player, you may do something like getPartyOfPlayer(x), and gets your own Party class (and I cannot handle that one). You may return an interface for the Party itself, so I can return my Parties implementation of that interface and it would works with your plugin too.
> 
> This would allows other plugins to hooks and handles parties for your plugin, otherwise directly hooks into other party plugins or Unite (a bridge for some party plugins).

Thanks for clarify this. I will update you both once I am free.

**Comment by liexes** [comment]
Good

**Comment by liexes** [comment]
@GoodestEnglish This has already been added?

**Comment by GoodestEnglish** [comment]
> @GoodestEnglish This has already been added?

Nope. This will most likely work on after most of the enhancement.

**Comment by Chromaah** [comment]
+1

I also want this :)

**Comment by GoodestEnglish** [comment]
Just saying, but this will not going to be done soon. There's much more important stuff which needs to be done first.

**Comment by MightCly** [comment]
+1
This is very important for my server, please add.
